test/parallel_api/algorithm/alg.nonmodifying/adjacent_find.pass.cpp

```diff
@@ -69,10 +69,10 @@ test_adjacent_find_by_type()
             auto i = ::std::adjacent_find(in.cbegin(), in.cend(), ::std::equal_to<T>());
             EXPECT_TRUE(i == in.cbegin() + e, "::std::adjacent_find returned wrong result");
 
-            invoke_on_all_policies<0>()(test_adjacent_find<T>(), in.begin(), in.end(), ::std::equal_to<T>());
+            invoke_on_all_policies<0>()(test_adjacent_find<T>(), in.begin(), in.end());
             invoke_on_all_policies<1>()(test_adjacent_find_predicate<T>(), in.begin(), in.end(), ::std::equal_to<T>());
```

Comment:
Could you please replace `std::equal_to` with any lambda, e.g. `[](...){return x == y;}` to allow implementation call with slightly complicated functor?

Reply:
Andrey, what's different between `std::equal_to `(https://en.cppreference.com/w/cpp/utility/functional/equal_to) and `[](...){return x == y;}` with the semantic, and the code coverage perspective?
I think ones are absolutely the same.
The other case - if we want to check passing a custom predicate into an algorithm's implementation. In that case  we have to a predicate/lambda with a semantic, different from `std::equal_to ` semantic.

Reply:
I still don't understand, why we need it, but is `[](...){return x%2 == y%2}` good for this purpose?

Reply:
My suggestion was to test with anything that is not `std::equal_to`. It can be any lambda, functor. As I mentioned above `[](...){return x == y;}` is just an example, another lambda can be used.
Regarding `[](...){return x%2 == y%2}`. It can also be used if it's applicable

Reply:
There is no difference between `std::equal_to `and `[](...){return x == y;}` with the test perspective here.
For better coverage  - yes, it make sense to use the other predicate, not `std::equal_to ` or the same by semantic. A sense here - no "using non-std predicate"...A sense is to use ANY predicate, with different from "equal" semantic, because the predicate with "equal" semantic we use within the implementation,  in version  of "adjacent_find without predicate". And we could call (in the implementation) such version w/o passing a custom predicate. That's case which we want to coverage.

Reply:
However,  a point about not-using `std::equal_to` in a test, probably also makes sense.. For example, If we have a specialization by `std::equal_to` in our implementation..

Reply:
Replaced `std::equal_to` with `[](...){ return (x - y) * (x - y) == 4; }` (distance between values == 2)